File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_18286.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request


async def BenchmarkTest18286(request: Request):
    path_value = request.path_params.get('id', '')
    data, _sep, _rest = str(path_value).partition('\x00')
    eval(str(data))
    return {"updated": True}

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest18286.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest18286 {

    private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(BenchmarkTest18286.class);

    @PostMapping(path="/BenchmarkTest18286", consumes="application/xml")
    public void BenchmarkTest18286(@RequestBody String xmlBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String xmlValue = xmlBody;
        String data = xmlValue.isEmpty() ? "default" : xmlValue;
        LOG.info("Action: {}", data);
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}
